src/middlewared/middlewared/plugins/ldap.py

```diff
@@ -28,6 +34,93 @@ class SSL(enum.Enum):
     USETLS = 'START_TLS'
 
 
+class NlscdConst(enum.Enum):
+    NSLCD_CONF_PATH = '/usr/local/etc/nslcd.conf'
+    NSLCD_PIDFILE = '/var/run/nslcd.pid'
+    NSLCD_SOCKET = '/var/run/nslcd/nslcd.ctl'
+    NSLCD_VERSION = 0x00000002
+    NSLCD_ACTION_STATE_GET = 0x00010002
+    NSLCD_RESULT_BEGIN = 1
+    NSLCD_RESULT_END = 2
+
+
+class NslcdClient(object):
+    def __init__(self, action):
+        # set up the socket (store in class to avoid closing it)
+        self.sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        fcntl.fcntl(self.sock, fcntl.F_SETFD, fcntl.FD_CLOEXEC)
+        # connect to nslcd
+        self.sock.connect(NlscdConst.NSLCD_SOCKET.value)
+        # self.sock.setblocking(1)
+        self.fp = os.fdopen(self.sock.fileno(), 'r+b', 0)
+        # write a request header with a request code
+        self.action = action
+        self.write_int32(NlscdConst.NSLCD_VERSION.value)
+        self.write_int32(action)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, typ, value, traceback):
+        self.close()
```

Comment:
Actually we should prob check for an exception and raising it after close

Reply:
Hm, I may be having a brainfart, I feel we talked about this before. Is the exception raised by default?